specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/2017-03-15-preview/OperationalInsights.json

```diff
@@ -0,0 +1,1320 @@
+        "name": {
+          "type": "string",
+          "description": "The name of the management group."
+        },
+        "id": {
+          "type": "string",
+          "description": "The unique ID of the management group."
+        },
+        "created": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The datetime that the management group was created."
+        },
+        "dataReceived": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The last datetime that the management group received data."
+        },
+        "version": {
+          "type": "string",
+          "description": "The version of System Center that is managing the management group."
+        },
+        "sku": {
+          "type": "string",
+          "description": "The SKU of System Center that is managing the management group."
+        }
+      },
+      "description": "Management group properties."
+    },
+    "ManagementGroup": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ManagementGroupProperties",
+          "description": "The properties of the management group."
+        }
+      },
+      "description": "A management group that is connected to a workspace"
+    },
+    "WorkspaceListManagementGroupsResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagementGroup"
+          },
+          "description": "Gets or sets a list of management groups attached to the workspace."
+        }
+      },
+      "description": "The list workspace managmement groups operation response."
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU.",
+          "enum": [
+            "Free",
+            "Standard",
+            "Premium",
+            "Unlimited",
+            "PerNode",
+            "Standalone"
+          ],
+          "x-ms-enum": {
+            "name": "SkuNameEnum",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The SKU (tier) of a workspace."
+    },
+    "Features": {
+      "properties": {
+        "legacy": {
+          "type": "string",
+          "description": "The legacy status of the workspace.",
+          "enum": [
+            "None",
+            "GrandfatheredNonSuite",
+            "GrandfatheredSuite"
+          ],
+          "x-ms-enum": {
+            "name": "legacyNameEnum",
+            "modelAsString": true
+          }
+        },
+        "searchVersion": {
+          "type": "string",
+          "description": "The search version of the workspace.",
+          "enum": [
+            "OQL",
+            "KQL"
+          ],
+          "x-ms-enum": {
+            "name": "searchVersionNameEnum",
```

Comment:
You probably don't need to name this `searchVersionNameEnum` -- `searchVersionName` would be ok, then generated classes don't need the `Enum` suffix in them.